tensorvis/tensorvisbase/include/inviwo/tensorvisbase/util/distancemetrics.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <vector>
+#include <inviwo/core/util/enumtraits.h>
+
+namespace inviwo {
+namespace tensorutil {
+enum class DistanceMetric { Euclidean, Manhattan, Minkowski, SquaredSum };
+
+// Input: Two vectors, each of which represent one point in an n-dimensional space
+// Output: Minkowski distance of order n between the two points
+template <typename T>
+T minkowskiDistance(const std::vector<T>& a, const std::vector<T>& b, const T order) {
+    if (a.size() != b.size())
```

Comment:
probably better to just IVW_ASSERT here since this is most likely a programming error and not something that can be fixed at runtime

Reply:
Conceptually, I agree, this should be an assertion. In practice, I'm not so sure, assertion does kill the application. Most of the time I hit assertions is while constructing new or modifying existing networks and then you want to be able to save your network even if an error has occurred.

Reply:
If I do that, I cannot run my unit tests anymore.

Reply:
we have auto save for network now, you can just click restore when you start Inviwo
Then remove those tests.

Reply:
Exception should not be used to catch code errors. If the response to seeing the exception is to change code, then it should not have been an exception.

Reply:
In this case the code might not be the problem but rather the data that you hand in.

Reply:
Yes, but that because the code that hands that data in is faulty